ios/X509RevocationChecker.swift

```diff
@@ -0,0 +1,140 @@
+  ) {
+    fetchCRL(from: crlURL, timeout: TimeInterval(readTimeout) / 1000.0) {
+      crlData, fetchError in
+      guard let crlData = crlData else {
+        completion(nil, fetchError)
+        return
+      }
+
+      // Guard against empty cert or crl (which would make baseAddress nil)
+      guard !certDER.isEmpty, !crlData.isEmpty else {
+        completion(nil, -2)
+        return
+      }
+
+      let result: Int32 = certDER.withUnsafeBytes { certBytes in
+        crlData.withUnsafeBytes { crlBytes in
+          guard let certPtr = certBytes.baseAddress,
+            let crlPtr = crlBytes.baseAddress
+          else {
+            return -998
+          }
+
+          if let issuer = issuerDER, !issuer.isEmpty {
+            return issuer.withUnsafeBytes { issuerBytes in
+              guard let issuerPtr = issuerBytes.baseAddress else {
+                return -998
+              }
+
+              return check_cert_revocation_with_crl(
+                certPtr.assumingMemoryBound(to: UInt8.self),
+                Int32(certDER.count),
+                crlPtr.assumingMemoryBound(to: UInt8.self),
+                Int32(crlData.count),
+                issuerPtr.assumingMemoryBound(to: UInt8.self),
+                Int32(issuer.count)
+              )
+            }
+          } else {
+            return check_cert_revocation_with_crl(
+              certPtr.assumingMemoryBound(to: UInt8.self),
+              Int32(certDER.count),
+              crlPtr.assumingMemoryBound(to: UInt8.self),
+              Int32(crlData.count),
+              nil, 0
+            )
+          }
+        }
+      }
+
+      switch result {
+      case 1:
+        completion(true, nil)
+      case 0:
+        completion(false, nil)
+      case -1, -7:
+        completion(nil, -1)  // .validationError
+      case -2:
+        completion(nil, -2)  // .crlParseFailed
+      case -3:
+        completion(nil, -3)  // .crlSignatureInvalid
+      case -4:
+        completion(nil, -4)  // .crlExpired
+      case -5:
+        completion(nil, -5)  // .validationError
+      case -6:
+        completion(nil, -6)  // .fetchFailed
+      default:
+        completion(nil, -999)  // Unknown/fallback error
+      }
+    }
+  }
+  /// Extracts the first CRL Distribution Point URI from a DER-encoded certificate.
+  /// Uses OpenSSL to parse the certificate and find the CRL URL.
+  ///
+  /// - Parameter certDER: The DER-encoded certificate from which to extract the CRL URI.
+  /// - Returns: A `URL` pointing to the CRL distribution point, or `nil` if not found or invalid.
+
+  public static func extractCRLDistributionPoint(from certDER: Data) -> URL? {
+    return certDER.withUnsafeBytes { ptr in
+      guard let base = ptr.baseAddress?.assumingMemoryBound(to: UInt8.self)
+      else { return nil }
+      guard let cString = extractCRLFromCert(base, Int32(certDER.count)) else {
+        return nil
+      }
+      return URL(string: String(cString: cString))
```

Comment:
This is where we need to pay attention: `extractCRLFromCert` states it's the caller's responsibility to free the URL memory, but we're currently not doing that. Am I correct?

I believe we need something like this:

```suggestion
      defer {
          free(cString)
      }
      return URL(string: String(cString: cString))
```